src/main/java/org/skriptlang/skript/bukkit/misc/rotation/NonMutatingQuaternionRotator.java

```diff
@@ -0,0 +1,49 @@
+package org.skriptlang.skript.bukkit.misc.rotation;
+
+import org.jetbrains.annotations.Contract;
+import org.joml.Quaternionf;
+import org.joml.Vector3f;
+
+import java.util.function.Function;
+
+public class NonMutatingQuaternionRotator implements Rotator<Quaternionf> {
+
+	private final Function<Quaternionf, Quaternionf> rotator;
+
+	/*
+	 * NOTE: the apparent mismatch between the axis and methods for local/non-local is intentional.
+	 * Rotating quaternions via rotateLocal results in a visual rotation around the global axis.
```

Comment:
is there a reason for this? just curious

Reply:
I'm really not sure. My guess is it's a difference in terminology usage.